composer/models/base.py

```diff
@@ -17,38 +18,125 @@
 
 
 class ComposerModel(torch.nn.Module, abc.ABC):
-    """The minimal interface needed to use a model with :class:`composer.trainer.Trainer`."""
+    """The interface needed to use a pytorch model with :class:`~composer.trainer.trainer.Trainer`.
+
+    To create a :class:`~composer.trainer.trainer.Trainer` compatible model, subclass :class:`~composer.models.base.ComposerModel` and
+    implement :meth:`forward` and :meth:`loss`. For full functionality (logging and validation), implement :meth:`metrics`
+    and :meth:`validate`.
+
+    Minimal Example:
+
+    .. code-block:: python
+
+        import torchvision
+        import torch.nn.functional as F
+
+        from composer.models import ComposerModel
+
+        class ResNet18(ComposerModel):
+
+            def __init__(self):
+                super().__init__()
+                self.model = torchvision.models.resnet18() # define pytorch model in __init__.
+
+            def forward(self, batch): # batch is the output of the dataloader
+                # specify how batches are passed through the model
+                inputs, _ = batch
+                return self.model(inputs)
+
+            def loss(self, outputs, batch):
+                # pass batches and `forward` outputs to the loss
+                _, targets = batch
+                return F.cross_entropy(outputs, targets)
```

Comment:
Should additionally add the same as above -> For more detail, see the composer model docs (with a :doc: link)